### PR TITLE
botonic improvements

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
Relevant commits:
* [renaming webview to webview-app](https://github.com/hubtype/botonic/pull/2401/commits/9094ddd6dcac9d2da970b4df87089f8be2159b11)
* [Adding TypeScript and rename webview-app.jsx to webview-app.tsx](https://github.com/hubtype/botonic/pull/2401/commits/3d097f0ff1663bb7d1955c254a32f2f42d97a174)
* [Convert Object.freeze to enums](https://github.com/hubtype/botonic/pull/2401/commits/066b195473b471669512350f7c377112899aaeb8)
* [Import only needed code in webview-app](https://github.com/hubtype/botonic/pull/2401/commits/2639993f2dd8dee768f31fd3b50b7a4d40751dc3) to facilitate tree-shaking with esbuild.

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

**Comment by asastre** [comment]
@vanbasten17 the tests are failing, can you take a look?

**Comment by vanbasten17** [comment]
> @vanbasten17 the tests are failing, can you take a look?

taking a look! 🆗

**Review by alexgimenez02 — COMMENTED** [review]
Doubt: In which node version was this PR developed? The workflow is running on node 12!

**Comment by alexgimenez02** [comment]
bt: In which node version was this PR developed? The workflow is running on node 12!

File: .github/workflows/botonic-react-tests.yml [hunk 1]
```diff
@@ -15,35 +15,35 @@ jobs:
...
-    - name: Verify lint
-      run: (cd ./packages/$PACKAGE && npm run lint_ci)
-    - name: Verify index.d.ts
-      run: scripts/qa/lint-d-ts.sh ./packages/$PACKAGE
+      - name: Checking out to current branch
+        uses: actions/checkout@v2
+      - name: Setting up node
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: '14'
```
Comment (alexgimenez02) [inline .github/workflows/botonic-react-tests.yml:23]: This solution of updating each version individually should be temporal, in the future I would like to create a reusable workflow where we just change for 1 file and all workflows update with it

**Review by alexgimenez02 — APPROVED** [review]

File: .github/workflows/botonic-react-tests.yml (hunk 1 above)
Comment (vanbasten17) [reply .github/workflows/botonic-react-tests.yml:23]: yes, we will keep improving it 😉

**Pull request merged**